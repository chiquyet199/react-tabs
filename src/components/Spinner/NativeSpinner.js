import React, {Component} from "react";
import {Modal} from "react-native";
import {
  Container,
  TextContainer,
  TextContent,
  StyledIndicator,
} from "./Spinner.style";
import {colors} from "../../constants/theme";

export default class Spinner extends Component {
  state = {
    visible: this.props.visible,
    textContent: this.props.textContent,
  };

  static defaultProps = {
    visible: false,
    cancelable: false,
    textContent: "",
    animation: "none",
    color: colors.white,
    size: "large",
    overlayColor: colors.semiTransparent,
  };

  close = () => {
    this.setState({visible: false});
  };

  componentWillReceiveProps = nextProps => {
    const {visible, textContent} = nextProps;
    this.setState({visible, textContent});
  };

  handleOnRequestClose = () => {
    if (this.props.cancelable) {
      this.close();
    }
  };

  _renderDefaultContent = () => {
    return (
      <Container>
        <StyledIndicator color={this.props.color} size={this.props.size} />
        <TextContainer>
          <TextContent>{this.state.textContent}</TextContent>
        </TextContainer>
      </Container>
    );
  };

  _renderSpinner = () => {
    const {visible} = this.state;
    if (!visible) {
      return null;
    }

    const spinner = (
      <Container
        overlaycolor={this.props.overlayColor}
        key={`spinner_${Date.now()}`}
      >
        {this.props.children
          ? this.props.children
          : this._renderDefaultContent()}
      </Container>
    );

    return (
      <Modal
        animationType={this.props.animation}
        onRequestClose={this.handleOnRequestClose}
        supportedOrientations={["landscape", "portrait"]}
        transparent
        visible={visible}
      >
        {spinner}
      </Modal>
    );
  };

  componentWillUnmount = () => {
    this.setState({visible: false});
  };

  render() {
    return this._renderSpinner();
  }
}
