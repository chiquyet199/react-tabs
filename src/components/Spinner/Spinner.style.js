import styled from "styled-components/native";

export const Container = styled.View`
  flex: 1;
  background-color: transparent;
  position: absolute;
  z-index: 1000;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: ${({theme, overlaycolor}) =>
    overlaycolor || theme.colors.none};
`;

export const StyledIndicator = styled.ActivityIndicator`
  flex: 1;
`;

export const TextContainer = styled.View`
  flex: 1;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  justify-content: center;
  align-items: center;
  position: absolute;
`;

export const TextContent = styled.Text`
  margin-top: 130;
  height: 50;
  font-size: 20;
  font-weight: bold;
  ${props => props.textStyle};
`;
