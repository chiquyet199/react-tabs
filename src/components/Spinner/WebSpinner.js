import React, {Component} from "react";
import {
  Container,
  TextContainer,
  TextContent,
  StyledIndicator,
} from "./Spinner.style";
import {colors} from "../../constants/theme";

export default class Spinner extends Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: this.props.visible,
      textContent: this.props.textContent,
    };
  }

  static defaultProps = {
    visible: false,
    cancelable: false,
    textContent: "",
    animation: "none",
    color: colors.white,
    size: "large",
    overlayColor: colors.semiTransparent,
  };

  close() {
    this.setState({visible: false});
  }

  componentWillReceiveProps(nextProps) {
    const {visible, textContent} = nextProps;
    this.setState({visible, textContent});
  }

  _renderDefaultContent() {
    return (
      <Container>
        <StyledIndicator color={this.props.color} size={this.props.size} />
        <TextContainer>
          <TextContent>{this.state.textContent}</TextContent>
        </TextContainer>
      </Container>
    );
  }

  _renderSpinner() {
    const {visible} = this.state;
    if (!visible) {
      return null;
    }

    return (
      <Container
        overlaycolor={this.props.overlayColor}
        key={`spinner_${Date.now()}`}
      >
        {this.props.children
          ? this.props.children
          : this._renderDefaultContent()}
      </Container>
    );
  }

  render() {
    return this._renderSpinner();
  }
}
