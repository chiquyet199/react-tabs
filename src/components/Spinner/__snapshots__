/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/Spinner/__native_snapshots__