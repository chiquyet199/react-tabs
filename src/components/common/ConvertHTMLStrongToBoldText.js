import React from "react";
import {Text} from "react-native";
import styled from "styled-components/native";

export const StyledView = styled(Text)`
  display: inline;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
`;

export default ({htmlString}) => {
  const parsed = [];
  let cursor = 0;
  let searchingFor = "<strong>";
  while (cursor < htmlString.length) {
    let endPoint = htmlString.indexOf(searchingFor, cursor);
    endPoint = endPoint === -1 ? htmlString.length : endPoint;
    const token = htmlString.substring(cursor, endPoint);
    if (searchingFor === "<strong>") {
      parsed.push(<Text key={token}>{token}</Text>);
    } else {
      // Styled-components seems to ignore font-weight here
      parsed.push(
        <Text style={{fontWeight: "bold"}} key={token}>
          {token}
        </Text>,
      );
    }
    cursor = endPoint + searchingFor.length;
    searchingFor = searchingFor === "<strong>" ? "</strong>" : "<strong>";
  }

  return <StyledView>{parsed}</StyledView>;
};
