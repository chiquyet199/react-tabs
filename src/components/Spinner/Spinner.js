import React from "react";
import {string, bool, oneOf} from "prop-types";
import NativeSpinner from "./NativeSpinner";
import {themed} from "../../utils/common";

const ANIMATION = ["none", "slide", "fade"];
const SIZES = ["small", "normal", "large"];

export const Spinner = props => <NativeSpinner {...props} />;

Spinner.propTypes = {
  visible: bool,
  cancelable: bool,
  textContent: string,
  color: string,
  animation: oneOf(ANIMATION),
  size: oneOf(SIZES),
  overlayColor: string,
};

export default themed(Spinner);
