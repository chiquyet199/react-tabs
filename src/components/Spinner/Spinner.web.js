import React from "react";
import {string, bool, oneOf} from "prop-types";
import WebSpinner from "./WebSpinner";
import {themed} from "../../utils/common";

const ANIMATION = ["none", "slide", "fade"];
const SIZES = ["small", "normal", "large"];

export const Spinner = props => <WebSpinner {...props} />;

Spinner.propTypes = {
  visible: bool.isRequired,
  cancelable: bool,
  textContent: string,
  color: string,
  animation: oneOf(ANIMATION),
  size: oneOf(SIZES),
  overlayColor: string,
};

export default themed(Spinner);
