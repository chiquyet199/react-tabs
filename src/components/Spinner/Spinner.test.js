import {Platform, Modal} from "react-native";
import React from "react";
import {shallow} from "enzyme";
import {Spinner} from "./Spinner";

import {TextContent, Container} from "./Spinner.style";
import configureStore from "redux-mock-store";
import theme, {colors} from "../../constants/theme";
import {isWeb, isNative} from "../../utils/common";
import WebSpinner from "./WebSpinner";
import NativeSpinner from "./NativeSpinner";
import {TextParagraph} from "../common/layout";

const text = "TEST";

const mockProps = {
  visible: true,
  cancelable: true,
  textContent: text,
  animation: "none",
  color: colors.white,
  size: "large",
  overlayColor: colors.semiTransparent,
  theme,
};

const mockPropsHidden = {
  visible: false,
  cancelable: false,
  textContent: text,
  animation: "none",
  color: colors.white,
  size: "large",
  overlayColor: colors.semiTransparent,
  theme,
};

const mockStore = configureStore(mockProps);

describe("Spinner component", () => {
  it("renders properly - " + Platform.OS, () => {
    const wrapper = shallow(<Spinner {...mockProps} />, {
      context: {store: mockStore(mockProps)},
    });
    expect(wrapper).toMatchSnapshot();
  });

  describe("Props Spinner testing", () => {
    it("test visibility", async () => {
      const shallowWrapper = shallow(<Spinner {...mockPropsHidden} />).dive();
      expect(shallowWrapper.get(0)).toBeFalsy();
    });
  });

  describe("visible tree functionality", () => {
    it("check _renderDefaultContent method", () => {
      const expectedType = isWeb ? WebSpinner : NativeSpinner;
      const shallowWrapper = shallow(<Spinner {...mockProps} />);
      const type = shallowWrapper.find(expectedType);
      expect(type).toHaveLength(1);
      expect(type.props().textContent).toEqual(text);
      expect(shallowWrapper.dive().find(TextContent)).toHaveLength(1);
    });
  });

  describe("passes children to Spinner", () => {
    it("check not _renderDefaultContent Container", () => {
      const expectedType = isWeb ? WebSpinner : NativeSpinner;
      const shallowWrapper = shallow(<Spinner {...mockProps} />);
      const type = shallowWrapper.find(expectedType);
      expect(type).toHaveLength(1);
      expect(shallowWrapper.dive().find(Container)).toHaveLength(2);
    });

    it("check not _renderDefaultContent TextParagraph", () => {
      const expectedType = isWeb ? WebSpinner : NativeSpinner;
      const shallowWrapper = shallow(
        <Spinner {...mockProps}>
          <TextParagraph theme={theme} globaltheme="light">
            {text}
          </TextParagraph>
        </Spinner>,
      );
      const type = shallowWrapper.find(expectedType);
      expect(type).toHaveLength(1);
      expect(shallowWrapper.dive().find(TextParagraph)).toHaveLength(1);
      expect(
        shallowWrapper
          .dive()
          .find(TextParagraph)
          .props().children,
      ).toEqual(text);
    });
    if (isNative) {
      it("check onRequestClose", () => {
        const shallowWrapper = shallow(<Spinner {...mockProps} />).dive();
        const onClose = shallowWrapper.find(Modal).prop("onRequestClose");
        onClose();
        shallowWrapper.update();
        expect(shallowWrapper.state("visible")).toBeFalsy();
        expect(shallowWrapper.find(Modal)).toHaveLength(0);
      });
    }
  });
});
