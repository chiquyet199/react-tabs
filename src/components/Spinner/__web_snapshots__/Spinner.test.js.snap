// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component renders properly - web 1`] = `
<Spinner
  animation="none"
  cancelable={true}
  color="#ffffff"
  overlayColor="rgba(0, 0, 0, 0.12)"
  size="large"
  textContent="TEST"
  theme={
    Object {
      "hq": Object {
        "buttons": Object {
          "confirmation_cancel": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "confirmation_danger": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#e26e59",
          },
          "confirmation_primary": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "danger": Object {
            "background": "#e26e59",
            "border": "#e26e59",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "danger_outlined": Object {
            "background": "#ffffff",
            "border": "#e26e59",
            "fontWeight": "bold",
            "text": "#e26e59",
          },
          "delete_button": Object {
            "background": "#ffffff",
            "border": "#b9c7d6",
            "color": "#b9c7d6",
            "fontWeight": "bold",
          },
          "facebook": Object {
            "background": "#3b5998",
            "border": "#3b5998",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "outlined": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "primary": Object {
            "background": "#5ab1b8",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "primary_outlined": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "#5ab88e",
            "fontWeight": "bold",
            "text": "#5ab88e",
          },
          "secondary": Object {
            "background": "#ffffff",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "site_page_util_bar_outlined": Object {
            "background": "#ffffff",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "site_page_util_bar_primary": Object {
            "background": "#5ab1b8",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "tabButton": Object {
            "background": "#ffffff",
            "border": "#e3eaf2",
            "fontSize": "14px",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "tertiary": Object {
            "background": "#ffffff",
            "border": "#6e7c91",
            "fontWeight": "bold",
            "text": "#6e7c91",
          },
          "transparent": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "transparent_blue": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "transparent_gray": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "transparent_green": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "14px",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "transparent_white": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
        },
        "colors": Object {
          "alertRed": "#f16651",
          "black": "#000000",
          "blueGray": "#455363",
          "blueGreen": "#3c7087",
          "canopyGreen": "#5ab88e",
          "canopySoftSteel": "#8999ab",
          "canopySoftSteelOverlay": "#f3f7fc",
          "canopySteel": "#6b7c93",
          "dark": "#010202",
          "darkConcrete": "#455363",
          "darkGreen": "#236a6d",
          "darkOverlay": "rgba(0, 0, 0, 0.8)",
          "darkTransparent": "rgba(0, 0, 0, 0.7)",
          "darkestGreen": "#011e24",
          "facebook": "#3b5998",
          "green": "#398d7f",
          "grey": "#6e7c91",
          "hqBlue": "#5ab1b8",
          "lightBackground": "rgba(255, 255, 255, 0.2)",
          "lightBlue": "#42c8ce",
          "lightGreen": "#58dfa3",
          "lightSteel": "#e3eaf2",
          "lightTeal": "#37b3ba",
          "none": "rgba(0, 0, 0, 0)",
          "orange": "#f1bf51",
          "primary": "#5ab88e",
          "red": "#e26e59",
          "rentPassportErrorBg": "#fff5f5",
          "rentPassportWarningBg": "#fffcf3",
          "semiTransparent": "rgba(0, 0, 0, 0.12)",
          "shadowColor": "rgba(107, 124, 147, 0.4)",
          "steel": "#b9c7d6",
          "successGreen": "#5dc471",
          "veryLightSteel": "#f6f9fc",
          "warning": "#fffcf3",
          "white": "#ffffff",
          "whiteTransparent": "rgba(255, 255, 255, 0.2)",
        },
        "dark": Object {
          "accentColor": "#5ab1b8",
          "activeTabColor": "#5ab1b8",
          "background": Array [
            "#ffffff",
            "#ffffff",
          ],
          "baseColor": "#ffffff",
          "dividerColor": "#e3eaf2",
          "form": Object {
            "accentColor": "#5ab1b8",
            "baseColor": "#6e7c91",
          },
          "inactiveTabColor": "#e3eaf2",
          "lineBreakColor": "#e3eaf2",
          "switch": Object {
            "accentColor": "#5dc471",
            "baseColor": "#ffffff",
          },
          "textColor": "#ffffff",
          "topbar": Object {
            "background": "rgba(0, 0, 0, 0)",
            "baseColor": "#ffffff",
            "headerButtonColour": "transparent_blue",
            "progressBar": "#42c8ce",
          },
        },
        "light": Object {
          "accentColor": "#5ab1b8",
          "activeTabColor": "#5ab1b8",
          "altTextColor": "#ffffff",
          "background": Array [
            "#ffffff",
            "#f6f9fc",
          ],
          "baseColor": "#6b7c93",
          "boxBorder": "#e3eaf2",
          "dividerColor": "#e3eaf2",
          "errorColor": "#e26e59",
          "form": Object {
            "accentColor": "#5ab1b8",
            "baseColor": "#6e7c91",
          },
          "inactiveTabColor": "#e3eaf2",
          "lineBreakColor": "#e3eaf2",
          "switch": Object {
            "accentColor": "#5dc471",
            "baseColor": "#ffffff",
          },
          "tabs": Object {
            "activeText": "#5ab1b8",
            "background": "#ffffff",
            "borderBottom": "#5ab1b8",
            "inactiveText": "#6b7c93",
            "indicator": "#5ab1b8",
            "label": "#5ab1b8",
          },
          "textColor": "#6b7c93",
          "topbar": Object {
            "background": "#ffffff",
            "baseColor": "#6b7c93",
            "headerButtonColour": "transparent_blue",
            "progressBar": "#42c8ce",
          },
        },
        "padding": Object {
          "top": 0,
        },
      },
      "hqApp": Object {
        "buttons": Object {
          "confirmation_cancel": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "confirmation_danger": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#e26e59",
          },
          "confirmation_primary": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "danger": Object {
            "background": "#e26e59",
            "border": "#e26e59",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "danger_outlined": Object {
            "background": "#ffffff",
            "border": "#e26e59",
            "fontWeight": "bold",
            "text": "#e26e59",
          },
          "delete_button": Object {
            "background": "#ffffff",
            "border": "#b9c7d6",
            "color": "#b9c7d6",
            "fontWeight": "bold",
          },
          "facebook": Object {
            "background": "#3b5998",
            "border": "#3b5998",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "outlined": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "primary": Object {
            "background": "#5ab1b8",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "primary_outlined": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "secondary": Object {
            "background": "#ffffff",
            "border": "#5ab1b8",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "tabButton": Object {
            "background": "#ffffff",
            "border": "#e3eaf2",
            "fontSize": "14px",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "tertiary": Object {
            "background": "#ffffff",
            "border": "#6e7c91",
            "fontWeight": "bold",
            "text": "#6e7c91",
          },
          "transparent": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "transparent_gray": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "transparent_green": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "14px",
            "fontWeight": "bold",
            "text": "#5ab88e",
          },
          "transparent_white": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
        },
        "colors": Object {
          "alertRed": "#f16651",
          "black": "#000000",
          "blueGray": "#455363",
          "blueGreen": "#3c7087",
          "canopyGreen": "#5ab88e",
          "canopySoftSteel": "#8999ab",
          "canopySoftSteelOverlay": "#f3f7fc",
          "canopySteel": "#6b7c93",
          "dark": "#010202",
          "darkConcrete": "#455363",
          "darkGreen": "#236a6d",
          "darkOverlay": "rgba(0, 0, 0, 0.8)",
          "darkTransparent": "rgba(0, 0, 0, 0.7)",
          "darkestGreen": "#011e24",
          "facebook": "#3b5998",
          "green": "#398d7f",
          "grey": "#6e7c91",
          "hqBlue": "#5ab1b8",
          "lightBackground": "rgba(255, 255, 255, 0.2)",
          "lightBlue": "#42c8ce",
          "lightGreen": "#58dfa3",
          "lightSteel": "#e3eaf2",
          "lightTeal": "#37b3ba",
          "none": "rgba(0, 0, 0, 0)",
          "orange": "#f1bf51",
          "primary": "#5ab88e",
          "red": "#e26e59",
          "rentPassportErrorBg": "#fff5f5",
          "rentPassportWarningBg": "#fffcf3",
          "semiTransparent": "rgba(0, 0, 0, 0.12)",
          "shadowColor": "rgba(107, 124, 147, 0.4)",
          "steel": "#b9c7d6",
          "successGreen": "#5dc471",
          "veryLightSteel": "#f6f9fc",
          "warning": "#fffcf3",
          "white": "#ffffff",
          "whiteTransparent": "rgba(255, 255, 255, 0.2)",
        },
        "dark": Object {
          "accentColor": "#5ab1b8",
          "activeTab": "#5ab88e",
          "altTextColor": "#ffffff",
          "background": Array [
            "#455363",
            "#455363",
          ],
          "baseColor": "#ffffff",
          "boxBorder": "#e3eaf2",
          "disabledTextColor": "#b9c7d6",
          "form": Object {
            "accentColor": "#5ab1b8",
            "baseColor": "#6e7c91",
          },
          "inactiveTab": "#e3eaf2",
          "lightBackground": "#f6f9fc",
          "switch": Object {
            "accentColor": "#5dc471",
            "baseColor": "#ffffff",
          },
          "textColor": "#ffffff",
          "topbar": Object {
            "background": "#455363",
            "baseColor": "#ffffff",
            "fontSize": 18,
            "headerButtonColour": "transparent_white",
          },
        },
        "light": Object {
          "accentColor": "#5ab1b8",
          "activeTab": "#5ab88e",
          "activeTenancy": Object {
            "propertySummaryBackground": "rgba(255, 255, 255, 0.2)",
          },
          "altBoxBorder": "#f6f9fc",
          "altTextColor": "#ffffff",
          "background": Array [
            "#ffffff",
            "#ffffff",
          ],
          "baseColor": "#6b7c93",
          "boxBorder": "#e3eaf2",
          "dangerText": "#e26e59",
          "disabledTextColor": "#b9c7d6",
          "dividerColor": "#e3eaf2",
          "errorColor": "#e26e59",
          "form": Object {
            "accentColor": "#5ab1b8",
            "baseColor": "#6e7c91",
          },
          "inactiveTab": "#e3eaf2",
          "lightBackground": "#f6f9fc",
          "lineBreakColor": "#6b7c93",
          "rentStatusTab": Object {
            "green": "#5dc471",
            "red": "#e26e59",
            "yellow": "#f1bf51",
          },
          "selectedBoxBorder": "#42c8ce",
          "slideoutMenu": Object {
            "background": "#3c7087",
          },
          "successText": "#5dc471",
          "switch": Object {
            "accentColor": "#5dc471",
            "baseColor": "#ffffff",
          },
          "tabs": Object {
            "activeText": "#5ab1b8",
            "background": "#ffffff",
            "borderBottom": "#5ab1b8",
            "inactiveText": "#6b7c93",
            "indicator": "#5ab1b8",
            "label": "#5ab1b8",
          },
          "textColor": "#6b7c93",
          "topbar": Object {
            "background": "#455363",
            "baseColor": "#ffffff",
            "fontSize": 18,
            "headerButtonColour": "transparent_white",
            "progressBar": "#42c8ce",
          },
        },
        "padding": Object {
          "top": 0,
        },
      },
      "selva": Object {
        "buttons": Object {
          "confirmation_cancel": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "confirmation_danger": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#e26e59",
          },
          "confirmation_primary": Object {
            "background": "#ffffff",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "18px",
            "fontWeight": "bold",
            "text": "#5ab1b8",
          },
          "danger": Object {
            "background": "#e26e59",
            "border": "#e26e59",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "danger_outlined": Object {
            "background": "#ffffff",
            "border": "#e26e59",
            "fontWeight": "bold",
            "text": "#e26e59",
          },
          "delete_button": Object {
            "background": "#ffffff",
            "border": "#b9c7d6",
            "color": "#b9c7d6",
            "fontWeight": "bold",
          },
          "facebook": Object {
            "background": "#3b5998",
            "border": "#3b5998",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "primary": Object {
            "background": "#5ab88e",
            "border": "#5ab88e",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "primary_outlined": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "#5ab88e",
            "fontWeight": "bold",
            "text": "#5ab88e",
          },
          "secondary": Object {
            "background": "#ffffff",
            "border": "#5ab88e",
            "fontWeight": "bold",
            "text": "#5ab88e",
          },
          "tabButton": Object {
            "background": "#ffffff",
            "border": "#e3eaf2",
            "fontSize": "14px",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "tertiary": Object {
            "background": "#ffffff",
            "border": "#6e7c91",
            "fontWeight": "bold",
            "text": "#6e7c91",
          },
          "transparent": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
          "transparent_gray": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#6b7c93",
          },
          "transparent_green": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontSize": "14px",
            "fontWeight": "bold",
            "text": "#5ab88e",
          },
          "transparent_white": Object {
            "background": "rgba(0, 0, 0, 0)",
            "border": "rgba(0, 0, 0, 0)",
            "fontWeight": "bold",
            "text": "#ffffff",
          },
        },
        "colors": Object {
          "alertRed": "#f16651",
          "black": "#000000",
          "blueGray": "#455363",
          "blueGreen": "#3c7087",
          "canopyGreen": "#5ab88e",
          "canopySoftSteel": "#8999ab",
          "canopySoftSteelOverlay": "#f3f7fc",
          "canopySteel": "#6b7c93",
          "dark": "#010202",
          "darkConcrete": "#455363",
          "darkGreen": "#236a6d",
          "darkOverlay": "rgba(0, 0, 0, 0.8)",
          "darkTransparent": "rgba(0, 0, 0, 0.7)",
          "darkestGreen": "#011e24",
          "facebook": "#3b5998",
          "green": "#398d7f",
          "grey": "#6e7c91",
          "hqBlue": "#5ab1b8",
          "lightBackground": "rgba(255, 255, 255, 0.2)",
          "lightBlue": "#42c8ce",
          "lightGreen": "#58dfa3",
          "lightSteel": "#e3eaf2",
          "lightTeal": "#37b3ba",
          "none": "rgba(0, 0, 0, 0)",
          "orange": "#f1bf51",
          "primary": "#5ab88e",
          "red": "#e26e59",
          "rentPassportErrorBg": "#fff5f5",
          "rentPassportWarningBg": "#fffcf3",
          "semiTransparent": "rgba(0, 0, 0, 0.12)",
          "shadowColor": "rgba(107, 124, 147, 0.4)",
          "steel": "#b9c7d6",
          "successGreen": "#5dc471",
          "veryLightSteel": "#f6f9fc",
          "warning": "#fffcf3",
          "white": "#ffffff",
          "whiteTransparent": "rgba(255, 255, 255, 0.2)",
        },
        "dark": Object {
          "accentColor": "#5ab88e",
          "background": Array [
            "#236a6d",
            "#011e24",
          ],
          "baseColor": "#f6f9fc",
          "dividerColor": "#e3eaf2",
          "form": Object {
            "accentColor": "#5ab88e",
            "baseColor": "#ffffff",
            "bounceButton": "transparent",
          },
          "switch": Object {
            "accentColor": "#5dc471",
            "baseColor": "#ffffff",
          },
          "tabs": Object {
            "activeText": "#5ab88e",
            "background": "#ffffff",
            "borderBottom": "#5ab88e",
            "inactiveText": "#6b7c93",
            "indicator": "#5ab88e",
            "label": "#5ab88e",
          },
          "textColor": "#f6f9fc",
          "topbar": Object {
            "background": "rgba(0, 0, 0, 0)",
            "baseColor": "#ffffff",
            "headerButtonColour": "transparent_white",
            "progressBar": "#58dfa3",
          },
        },
        "light": Object {
          "accentColor": "#5ab88e",
          "background": Array [
            "#ffffff",
            "#ffffff",
          ],
          "baseColor": "#6b7c93",
          "dividerColor": "#e3eaf2",
          "errorTextColor": "#f16651",
          "form": Object {
            "accentColor": "#5ab88e",
            "baseColor": "#6b7c93",
            "bounceButton": "transparent_green",
          },
          "switch": Object {
            "accentColor": "#5dc471",
            "baseColor": "#ffffff",
          },
          "tabs": Object {
            "activeText": "#5ab88e",
            "background": "#ffffff",
            "borderBottom": "#5ab88e",
            "inactiveText": "#6b7c93",
            "indicator": "#5ab88e",
            "label": "#5ab88e",
          },
          "textColor": "#6b7c93",
          "topbar": Object {
            "background": "#ffffff",
            "baseColor": "#6b7c93",
            "fontSize": 18,
            "headerButtonColour": "transparent_green",
            "progressBar": "#58dfa3",
          },
        },
        "padding": Object {
          "top": 0,
        },
      },
    }
  }
  visible={true}
/>
`;
